Keyboard backlight handling improvements

cwendling opened the pull request:
See commit log, but most important bits:
* [**Fix toggling backlight back on**](https://github.com/mate-desktop/mate-power-manager/commit/1def59bb169006897048b8de351e1ddd1bba4441).  When triggering the toggle action, without this it turns off just fine, but never comes back: one has to use the up action enough times to get back the previous brightness.  This was a simple one.
* [**Persistently save user-set brightness**](https://github.com/mate-desktop/mate-power-manager/commit/4d725c390cd5f7f19d2aea8c7cddf88c89b51cc4).  Currently the brightness was never saved, so one would have had to manually change the "brightness on AC" value to restore a desired value.  This seems overly annoying when having up/down/toggle keyboard buttons that the values they set is not remembered.  So, I changed this so that user actions (e.g. button triggers) modify the saved value, and is used to restore next time the daemon starts.

The rest is cleanups, and [a small change to the handling of some settings combinations](https://github.com/mate-desktop/mate-power-manager/commit/80466432a003431c6c8319a3ad90a52b765fc082) because they were just odd in some corner cases, and mostly irrelevant now the user changes are preserved.

---

I *guess* that not many people use keyboard backlight under MATE, because it seems really under-loved and especially the toggle bug seems both annoying and simple enough that it would have been spotted and fixed before.  I myself just got my hands on a laptop with keyboard backlight with a proper Linux driver for controlling it requiring no hacks, and I couldn't help myself but to fix this :slightly_smiling_face: 
Hopefully I can still get some review on this, or just rely on street cred :v:

lukefromdc commented:
You will want to rebase this against master to reflect that the last commit in this PR has been effectivelhy merged separately to fix the Travis problems

lukefromdc reviewed (approved):
On my desktop with no backlit keyboard I've never seen any reference to a keyboard backlight control. Same on my laptops, none of which have ever had a backlit keyboard. Assuming this is shown in the power manager preferences only when the relevent hardware is detected.

That said, the build finished fine and nothing changed in a quick test running both mate-power-manager and mate-control-center with GDK_BACKEND=x11. This generates a ton of warnings in the terminal the power manager is running from but is enough to show the normal GUI in the preferences dialog.

cwendling commented:
I just rebased on top of fixed master.

@lukefromdc thanks for the quick test.  However, note that nothing I changed here is related to GUI preferences themselves, but solely to how hardware buttons react and how some settings combination interact.

I'd be best to have others test this with the relevant hardware, but I also believe that it was broken badly enough (IMO) that even if there's some more fine-tuning to do to please everyone it's still way better than the state before.

lukefromdc commented:
So presuming I should wait awhile to see if we can get another test, but if this just
sits with no further reviews merge it?

cwendling commented:
@lukefromdc yeah, I believe it's a good compromise.  And maybe it's been long enough that it makes sense to merge and see if anybody actually notices things changed :)

lukefromdc commented:
Merge is done, let's hope this helps. Again, I don't have the relevent hardware but this should not affect anyone without a backlit keyboard.
